include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -74,11 +74,18 @@ struct maybe_remove_property<PoolResource,
                              Upstream,
                              Property,
                              cuda::std::enable_if_t<!cuda::has_property<Upstream, Property>>> {
+#ifdef __GNUC__  // GCC has warns about compatibility issues with pre ISO C++ code
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnon-template-friend"
```

Comment:
Is this supposed to be `-Wno-non-template-friend`? Let's be sure this solves the problem before merging.
```suggestion
#pragma GCC diagnostic ignored "-Wno-non-template-friend"
```

Reply:
This needs to be `-Wnon-template-friend` The other is the comandline option that would silence the warning

See https://gcc.godbolt.org/z/jbWjK3o9j

This unfortunately does not work for gcc5 and below

Reply:
Okay, cool. 👍 Thanks!